opentelemetry-sdk/src/opentelemetry/sdk/_metrics/aggregation.py

```diff
@@ -16,82 +16,138 @@
 
-    def make_point_and_reset(self):
-        pass
+    def _collect(self):
+        now = _time_ns()
 
+        with self._lock:
+            self._value = 0
+            self._start_time_unix_nano = now + 1
 
-class LastValueAggregation(Aggregation):
+        return Sum(
+            aggregation_temporality=AGGREGATION_TEMPORALITY_DELTA,
+            is_monotonic=self._is_monotonic,
+            start_time_unix_nano=self._start_time_unix_nano,
+            time_unix_nano=now,
+            value=self._value,
+        )
 
-    """
-    This aggregation collects data for the SDK sum metric point.
-    """
 
-    def __init__(self, instrument):
-        self._value = None
-        self._timestamp = _time_ns()
+class AsynchronousSumAggregation(Aggregation[Sum]):
+    def __init__(self, is_monotonic: bool):
+        super().__init__(is_monotonic)
+        self._start_time_unix_nano = _time_ns()
 
-    def aggregate(self, value):
-        self._value = value
-        self._timestamp = _time_ns()
+    def aggregate(self, measurement: Measurement) -> None:
+        with self._lock:
+            self._value = measurement.value
 
-    def make_point_and_reset(self):
-        pass
+    def _collect(self):
+        if self._value is None:
+            return None
+
+        return Sum(
+            start_time_unix_nano=self._start_time_unix_nano,
+            time_unix_nano=_time_ns(),
+            value=self._value,
+            aggregation_temporality=AGGREGATION_TEMPORALITY_CUMULATIVE,
+            is_monotonic=self._is_monotonic,
+        )
 
 
-class ExplicitBucketHistogramAggregation(Aggregation):
+class LastValueAggregation(Aggregation[Gauge]):
+    def aggregate(self, measurement: Measurement):
+        with self._lock:
+            self._value = measurement.value
+
+    def _collect(self):
+        if self._value is None:
+            return None
+
+        return Gauge(
+            time_unix_nano=_time_ns(),
+            value=self._value,
+        )
 
-    """
-    This aggregation collects data for the SDK sum metric point.
-    """
 
+class ExplicitBucketHistogramAggregation(Aggregation[Histogram]):
     def __init__(
         self,
-        instrument,
-        *args,
-        boundaries=(0, 5, 10, 25, 50, 75, 100, 250, 500, 1000),
-        record_min_max=True,
+        is_monotonic: bool,
+        boundaries: Sequence[int] = (
+            0,
+            5,
+            10,
+            25,
+            50,
+            75,
+            100,
+            250,
+            500,
+            1000,
+        ),
+        record_min_max: bool = True,
     ):
-        super().__init__()
+        super().__init__(is_monotonic)
         self._value = OrderedDict([(key, 0) for key in (*boundaries, inf)])
         self._min = inf
         self._max = -inf
         self._sum = 0
-        self._instrument = instrument
         self._record_min_max = record_min_max
+        self._start_time_unix_nano = _time_ns()
 
     @property
     def min(self):
```

Comment:
these properties can be removed now i think

Reply:
All right, removed them